src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -127,6 +127,9 @@ internal void AbortRequest()
 
         internal async Task CompleteResponseAsync()
         {
+            // Flush any pending content in writer
+            await _httpContext.Response.BodyWriter.FlushAsync();
```

Comment:
Isn't this redundant with `PipeWriter.Complete()` that's called from `CompleteWrites()`? Do you just need to reverse the calls to ReturnResponseMessageAsync and CompleteWrites?

Reply:
Yeah. I'm not sure I understand this change. If the PipeWriter is completed, that should flush any pending content. The reader might see the pending data until after Complete() returns, but that's also true for FlushAsync() with a PauseWriterThreshold of anything other than 1.

Reply:
It isn't redundant because the unit test fails if FlushAsync isn't here.

Is there a better way to do this?